src/Sarif/ExtensionMethods.cs

```diff
@@ -209,7 +209,7 @@ public static string GetMessageText(this Result result, IRule rule, bool concise
             {
                 text = string.Empty;    // Ensure that it's not null.
 
-                if (rule != null)
+                if (rule != null && result.FormattedRuleMessage != null)
```

Comment:
An unrelated bug: you could get an NRE if there _was_ a rule object, but `result` did not have a `FormattedRuleMessage` property.

*sigh* I should add a test for this one too...

Reply:
Ok, I filed #646 for this, and added the unit test.

Reply:
Looks good

---
In reply to: [146093210](https://github.com/Microsoft/sarif-sdk/pull/646#discussion_r146093210) [](ancestors = 146093210)